test/performance/search/search_benchmark.cpp

```diff
@@ -0,0 +1,285 @@
+}
+
+template <Alphabet alphabet_t>
+void mutate_substitution(std::vector<alphabet_t> & seq, size_t const overlap, size_t const seed)
+{
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<uint8_t> dis_alpha_short(0, alphabet_size<alphabet_t> - 2);
+    std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(seq) - overlap);
+    alphabet_t & cbase = seq[random_pos(gen)];
+    uint8_t crank = to_rank(cbase);
+    uint8_t rrank = dis_alpha_short(gen);
+    if (rrank >=  crank)
+        ++rrank;
+    cbase.assign_rank(rrank);
+}
+
+template <Alphabet alphabet_t>
+auto generate_reads(std::vector<alphabet_t> & ref,
+                    size_t const number_of_reads,
+                    size_t const read_length,
+                    float const simulated_errors_v,
+                    float const prob_insertion,
+                    float const prob_deletion,
+                    bool const error_dis = false,
+                    size_t const seed = 0)
+{
+    std::vector<std::vector<alphabet_t> > reads;
+    std::mt19937 gen(seed);
+    std::uniform_int_distribution<size_t> seeds (0, SIZE_MAX);
+    std::normal_distribution<> dis_error_count{0, simulated_errors_v};
+    std::uniform_real_distribution<double> probability(0.0, 1.0);
+
+    for (size_t i = 0; i < number_of_reads; ++i)
+    {
+        // TODO avoid mutating the same position multiple times
+        int simulated_errors;
+        // simulate concrete error number or use normal distribution 
+        if(!error_dis)
+            simulated_errors = std::round(simulated_errors_v);
+        else
+            simulated_errors = abs(std::round(dis_error_count(gen)));
+
+        std::uniform_int_distribution<size_t> random_pos(0, std::ranges::size(ref) - read_length - simulated_errors);
+        size_t rpos = random_pos(gen);
+        std::vector<alphabet_t> read_tmp{ref.begin() + rpos,
+            ref.begin() + rpos + read_length + simulated_errors};
+        for (int j = 0; j < simulated_errors; ++j)
+        {
+            double prob = probability(gen);
+            // Substitution
+            if (prob_insertion + prob_deletion < prob)
+            {
+                mutate_substitution(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Insertion
+            else if (prob_insertion < prob)
+            {
+                mutate_insertion(read_tmp, simulated_errors, seeds(gen));
+            }
+            // Deletion
+            else
+            {
+                mutate_deletion(read_tmp, simulated_errors, seeds(gen));
+            }
+        }
+        read_tmp.erase(read_tmp.begin() + read_length, read_tmp.end());
+        reads.push_back(read_tmp);
+    }
+    return reads;
+}
+
+template <typename alphabet_t>
+auto generate_repeating_sequence(size_t const template_length = 5000,
+                                 size_t const repeats = 20,
+                                 double const repeats_length = 1,
```

Comment:
```suggestion
                                 size_t const repeats_length = 1,
```

Reply:
repeats_length renamed to template_fraction. This parameter discribes the size (in releation to template_length (0-1)) of template which is copied in each iteration.